api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -388,4 +510,232 @@ default void deleteFile( @NonNull GenericFilePath path ) throws OperationFailedE
+   * <p>
+   * The default implementation of this method parses the given path's string representation using
+   * {@link GenericFilePath#parseRequired(String)} and then calls {@link #renameFile(GenericFilePath, String)}
+   * with the result.
+   *
+   * @param path    The string representation of the path of the file to be renamed. This path must not correspond to
+   *                a file in the trash/deleted.
+   * @param newName The new name of the file. This name must not be empty, and must not contain any control characters.
+   * @return {@code true} if the file was renamed, {@code false} if the file already had the new name.
+   * @throws AccessControlException   If the current user cannot perform this operation.
+   * @throws InvalidPathException     If the specified path is not valid, or if the specified path's string
+   *                                  representation is not valid, according to
+   *                                  {@link GenericFilePath#parseRequired(String)}.
+   * @throws NotFoundException        If the specified path does not exist, or does correspond to a file in the
+   *                                  trash/deleted, or the current user is not allowed to access it.
+   * @throws ConflictException        If the path with the new name already exists.
+   * @throws OperationFailedException If the operation fails for some other (checked) reason.
+   */
+  default boolean renameFile( @NonNull String path, @NonNull String newName ) throws OperationFailedException {
+    return renameFile( GenericFilePath.parseRequired( path ), newName );
+  }
+
+  /**
+   * Copies files, given their paths, to a destination path.
+   *
+   * @param paths           The list of file paths to be copied. These paths must not correspond to files that are in
+   *                        the trash/deleted.
+   * @param destinationPath The file path to copy the files to.
+   * @throws AccessControlException        If the current user cannot perform this operation.
+   * @throws BatchOperationFailedException If the batch operation fails for some reason.
+   * @throws OperationFailedException      If the operation fails for some other (checked) reason.
+   */
+  void copyFiles( @NonNull List<GenericFilePath> paths, @NonNull GenericFilePath destinationPath )
+    throws OperationFailedException;
+
+  /**
+   * Copies a file, given its path, to a destination path.
+   *
+   * @param path            The file path to be copied. This path must not correspond to a file in the trash/deleted.
+   * @param destinationPath The file path to copy the files to. This path must not correspond to a file in the
+   *                        trash/deleted.
+   * @throws AccessControlException    If the current user cannot perform this operation.
+   * @throws InvalidOperationException If the destination path is not valid.
+   * @throws NotFoundException         If either path does not exist or does correspond to a file in the
```

Comment:
Have another comment on this... it would be nice to be able to distinguish which path is not found. Browse Files API has the expressiveness for that.